.github/workflows/up_ex_doc_version.yml

```diff
@@ -0,0 +1,49 @@
+---
+name: up_ex_doc_version
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Run daily at midnight
+  workflow_dispatch:
+  
+
+jobs:
+  test:
+    name: Update ex_doc version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get current version
+        run: |
+            CURR=`curl -s https://hex.pm/api/packages/ex_doc | jq -r '.latest_stable_version' \
+                  jq -r ".[0].name"
+            TITLE="\`ex_doc\` ${CURR} released"
+            PR_EXISTS=`curl -s https://api.github.com/repos/starbelly/rebar3_ex_doc/pulls?state=all%26per_page=100 | grep -c "\"title\": \"${TITLE}\""`
+
+            if [ ${PR_EXISTS} -ne 0 ]
+            then
+              git config user.name "GitHub Actions"
+              git config user.email "actions@user.noreply.github.com"
+
+              BRANCH="up_ex_doc_to_${CURR}"
+
+              if git branch -a | grep "$BRANCH" > /dev/null; then
+                  # already exists
+                  exit
+              fi
+
+              git fetch origin
+              git checkout -b "$BRANCH"
+              mix compile
+              mix up_ex_doc_version "${CURR}"
+
+              if ! git diff --exit-code 1> /dev/null ; then
+                  # there's stuff to push
+                  git add .
+                  git commit -m "Update ex_doc version"
+                  git push origin "$BRANCH"
+
+                  gh pr create --fill \
+                      --title "Update ex_dpc version (automation)" \
+                      --body "This is an automated action to update ex_doc version"
+              fi
+            fi
```

Comment:
```suggestion
            fi
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Without this you can't push commits or create pull requests. More info. here: https://docs.github.com/en/actions/security-guides/automatic-token-authentication